source/templates.tex

```diff
@@ -1245,27 +1253,28 @@
 \rSec2[temp.arg.nontype]{Template non-type arguments}
 
 \pnum
-If the type of a \grammarterm{template-parameter}
-contains a placeholder type~(\ref{dcl.spec.auto}, \ref{temp.param}),
-the deduced parameter type is determined
-from the type of the \grammarterm{template-argument}
-by placeholder type deduction\iref{dcl.type.auto.deduct}.
+If the type \tcode{T} of a \grammarterm{template-parameter}\iref{temp.param}
+contains a placeholder type\iref{dcl.spec.auto}
+or a placeholder for a deduced class type\iref{dcl.type.class.deduct},
+the type of the parameter is the type deduced
+for the variable \tcode{x} in the invented declaration:
+\begin{codeblock}
+  T x = @\grammarterm{template-argument}@ ;
```

Comment:
Maybe try removing the italic corection after template-argument? There's more whitespace there than I'd prefer.

Reply:
That's a hard one. Fiddling with the italics correction doesn't change anything (same before and after):
![before](https://user-images.githubusercontent.com/23412755/41335079-e52d8810-6ee7-11e8-8bde-69fb28788048.png)

\kern to the rescue:
![kern](https://user-images.githubusercontent.com/23412755/41335481-3cfcfe08-6ee9-11e8-9fdd-c05ff41d4a33.png)

Reply:
`\itcorr[-1]` does help:

![screenshot from 2018-06-18 17-00-27](https://user-images.githubusercontent.com/232783/41568205-31cfb9ce-7319-11e8-90c4-ebd03de9de49.png)

That's:

```
  T x = @\grammarterm{template-argument}@ ;
  T x = @\grammarterm{template-argument}\itcorr[-1]@ ;
  T x = @\grammarterm{template-argument}\kern-0.3em@ ;
```

I'm going with the `\itcorr` approach for consistency with what we do elsewhere. I think it also makes the left/right whitespace look more balanced than the 0.3em negative kerning hint.

Reply:
@zygoloid Does the `\itcorr[-1]` give you the same result as using `\grammartermnc`?

Reply:
@godbyk It didn't, but that's because `\grammartermnc` was broken. Fixed in 1478e7c3.